crates/fluvio-connector-derive/src/generator.rs

```diff
@@ -45,17 +62,32 @@ fn generate_sink(func: &ConnectorFn) -> TokenStream {
     quote! {
 
         fn main() -> ::fluvio_connector_common::Result<()> {
-            let stop_signal = ::fluvio_connector_common::consumer::init_ctrlc()?;
             #init_and_parse_config
+            let stop_signal = ::fluvio_connector_common::consumer::init_ctrlc()?;
 
             ::fluvio_connector_common::future::run_block_on(async {
-                let (fluvio, stream) = ::fluvio_connector_common::consumer::consumer_stream_from_config(&common_config).await?;
+                let (fluvio, mut stream) = ::fluvio_connector_common::consumer::consumer_stream_from_config(&common_config).await?;
 
                 let metrics = ::std::sync::Arc::new(::fluvio_connector_common::monitoring::ConnectorMetrics::new(fluvio.metrics()));
                 ::fluvio_connector_common::monitoring::init_monitoring(metrics);
 
-                let mut stream = stream.take_until(stop_signal.recv());
-                #user_fn(user_config, stream).await
+                ::fluvio_connector_common::future::select! {
+                    user_fn_result = async {
+                        #user_fn(user_config, stream).await
+                    } => {
+                        match user_fn_result {
+                            Ok(_) => ::fluvio_connector_common::tracing::info!("Connector arrived at end of stream"),
+                            Err(e) => {
+                                ::fluvio_connector_common::tracing::error!(%e, "Connector failed");
```

Comment:
instead of `connector failed`,  say something like: `error encountered processing records in sink`. it would be good to add connector type?
Also try to add offset and other metadata that could be help to diagnose problem

Reply:
sure, wdyt "Error encountered producing records in source" for source?

Reply:
offset? 
I changed this part of code just to have sure that we're printing the error form connector.